tests/fixtures/tests.py

```diff
@@ -804,3 +804,19 @@ def test_forward_reference_m2m(self):
             t1.other_things.order_by('key'),
             ['<NaturalKeyThing: t2>', '<NaturalKeyThing: t3>']
         )
+
+
+class CircularReferenceTests(DumpDataAssertMixin, TestCase):
+    def test_circular_reference(self):
+        management.call_command('loaddata', 'circular_reference.json', verbosity=0)
+        obj_a = CircularA.objects.get()
+        obj_b = CircularB.objects.get()
+        self.assertEqual(obj_a.obj, obj_b)
+        self.assertEqual(obj_b.obj, obj_a)
+        self._dumpdata_assert(
+            ['fixtures'],
+            '[{"model": "fixtures.circulara", "pk": 1, '
+            '"fields": {"key": "x", "obj": 1}}, '
+            '{"model": "fixtures.circularb", "pk": 1, '
+            '"fields": {"key": "y", "obj": 1}}]',
```

Comment:
Why assert against a string here (and other places in this PR) and not against the json file as I originally had? (though looking back [at my code](https://github.com/django/django/commit/9d74f0975405d19daead0158275518d6228fd2b5#diff-14b4883f2cf50c9daf38f2b4ff9224acR849), I wonder if that worked at all, since `_dumpdata_assert` cannot the check file from file? Huh?)

Reply:
Yes, it worked because you added a special [hook](https://github.com/django/django/commit/9d74f0975405d19daead0158275518d6228fd2b5#diff-14b4883f2cf50c9daf38f2b4ff9224acR793-R796) for that, but I think it's more readable to assert with strings because these fixtures are quite small.

Reply:
Ah, I see. I looked for such a change in the original method, not in an override. Maybe that's a sign that such an override isn't the best approach in any case. Using a string might indeed be better, then :-)